src/main/java/at/hannibal2/skyhanni/features/misc/IslandLeaveJoinMsgs.kt

```diff
@@ -0,0 +1,145 @@
+
+        val joinMessage = " §${if (config.leaveJoinColor) "a" else "e"}joined §ethe island."
+        val leaveMessage = " §${if (config.leaveJoinColor) "c" else "e"}left §ethe island."
+
+        for (line in event.tabList) {
+            if (guesting) {
+                islandCategoryPattern.matchMatcher(line) {
+                    inIslandCategory = true
+                }
+                if (inIslandCategory) offlinePlayerPattern.matchMatcher(line) {
+                    islandOwners.add(group("player"))
+                }
+                guestCategoryPattern.matchMatcher(line) {
+                    inIslandCategory = false
+                    for (player in islandOwners) {
+                        if (isPlayerKnown(player.removeColor())) {
+                            onKnownIsland = true
+                            return@matchMatcher
+                        }
+                    }
+                    onKnownIsland = false
+                    return@matchMatcher
+                }
+            }
+
+            cleanPlayerPattern.matchMatcher(line) {
+                val player = group("player")
+                playersNew.add(player)
+                if (guesting && inIslandCategory) { islandOwners.add(player) }
+                if (players.contains(player)) {
+                    return@matchMatcher
+                }
+                players.add(player)            // !onPrivateIslandGarden because a vanilla message gets sent
+                if (shouldSendMsg(player) && updatedSinceWorldSwitch && !onPrivateIslandGarden) {
+                    ChatUtils.chat(player + joinMessage)
+                }
+            }
+            rawPlayerPattern.matchMatcher(line) {
+                val player = group("player")
+                if (!playersNew.contains(player)) playersNew.add(player)
+            }
+        }
+
+        if (players.isEmpty()) return
+
+        if (players.size > 1) {
+            updatedSinceWorldSwitch = true
+        }
+
+        for ((index, player) in players.withIndex().reversed()) {
+            if (!playersNew.contains(player)) {
+                if (shouldSendMsg(player)) ChatUtils.chat(player + leaveMessage)
+                players.removeAt(index)
+            }
+        }
+    }
+
+    @SubscribeEvent
+    fun onWorldChange(event: LorenzWorldChangeEvent) {
+        clearPlayers()
+    }
+
+    fun clearPlayers() {
+        players.clear()
+        updatedSinceWorldSwitch = false
+        onKnownIsland = false
+    }
+
+    private fun shouldSendMsg(player: String): Boolean {
+        val cleanPlayer = player.removeColor()
+
+        if (LorenzUtils.getPlayerName() == cleanPlayer) return false
+        if (!config.onlyKnownPeople) return true
+        return isPlayerKnown(cleanPlayer) || (config.alwaysOnYourIsland && IslandType.onPrivateWorld()) || (config.alwaysOnKnownIslands && onKnownIsland)
+    }
+
+    private fun isPlayerKnown(player: String): Boolean {
```

Comment:
this is duplicate to something in advanced tab list. write a common utils/api function instead

Reply:
https://discord.com/channels/997079228510117908/1094190239532208228/1213636854671474779

Reply:
"Is player known (name) = get social score (name) > 1" (pseudo code)